OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -124,7 +132,12 @@ Order IIssueDeployOrder.IssueDeployOrder(Actor self, bool queued)
 
 		bool IIssueDeployOrder.CanIssueDeployOrder(Actor self, bool queued)
 		{
-			return IsCellAcceptable(self, self.Location);
+			return CanLayMineOrRearm();
+		}
+
+		bool CanLayMineOrRearm()
+		{
+			return (HasSufficientAmmo() && IsCellAcceptable(self, self.Location)) || (!HasSufficientAmmo() && rearmableInfo != null);
```

Comment:
Shouldn't this still check if the cell is acceptable in all cases? Otherwise an order will be issued for a cell that is not eligible. (And the minelayer will return to that cell after rearming?)
```suggestion
			return IsCellAcceptable(self, self.Location) && (HasSufficientAmmo() || rearmableInfo != null);
```

Reply:
This would be fine if the check for actors in the same cell is removed (below), otherwise it'll prevent rearming when you're on a cell with a mine in it (after laying the last mine, which is when you'll tend to want to rearm).

Reply:
Yes, which is why I think we shouldn't offer a "lay mine" order here. Offering a "rearm" order is Imho a different issue. (Which involves artwork, as you highlighted in your comment above.)